contrib/platform/src/com/sun/jna/platform/win32/Psapi.java

```diff
@@ -320,4 +338,79 @@ class PERFORMANCE_INFORMATION extends Structure {
         public DWORD ProcessCount;
         public DWORD ThreadCount;
     }
+
+    @FieldOrder({"Flags"})
+    class PSAPI_WORKING_SET_EX_BLOCK extends Structure {
+        public ULONG_PTR[2] Flags = new ULONG_PTR[Native.POINTER_SIZE == 8 ? 1 : 2];
+        private Long innerValue;
+        private void setInnerValue() {
+            if (innerValue != null) {
+                return;
+            } else if (this.Flags.length == 2) {
+                Long upper = Optional.ofNullable(Flags[1]).map((elem) -> elem.longValue()).orElse(0l);
```

Comment:
I love `Optional`, but we are presently (at least as long as we're on 5.x) still maintaining JDK 6 compatibility.   But looking at the structure definition, the upper 32 bits are always unusued (`ReservedUlong`) so I think you're completely safe to ignore `Flags[1]` and simply write the 32 bits of `Flags[0]` to the innerValue.  (You could also use `int` if you're careful with the sign bit when shifting.)

Reply:
JDK 6 Compatibility. Now that is commitment, I'll adjust to handle JDK6